### PR TITLE
Improved localisation

### PR DESCRIPTION
**Pull request by mister-ben**
Replaces #1873, which got messy after being stagnant so long...

* Try primary code before default string, e.g. if there's no `de-DE` translation, use `de` if available. 
* Treat language codes case insensitively. Safari uses `en-us` but Chrome and Firefox use `en-US`, these should be treated the same.
* Re-merge global and player languages so that `addLanguage()` is still effective for players that had languages set as a setup option. Player-local languages take precedence.

### CODE REVIEW

File: src/js/player.js [hunk 1]
```diff
@@ -104,7 +104,17 @@ class Player extends Component {
...
-    this.languages_ = options['languages'] || Options['languages'];
+    if (options['languages']) {
+      // Normalise player option languages to lowercase
+      let languagesToLower = {};
+      Object.getOwnPropertyNames(options['languages']).forEach(function(value,index,array){
+        languagesToLower[value.toLowerCase()] = options['languages'][value];
+      });
+      this.languages_ = languagesToLower;
+    }
+    else {
```
Comment (dmlap) [inline src/js/player.js]: This `else` should be on the same line as the closing curly-brace of the preceding `if`.

File: docs/guides/languages.md [hunk 2]
```diff
@@ -122,6 +122,24 @@ During a Video.js player instantiation you can force it to localize to a specifi
 </video>
 ```
 
+Determining Player Language
```
Comment (dmlap) [inline docs/guides/languages.md]: Very nice addition.

**Comment by dmlap** [comment]
I think there may be an additional whitespace issue in here (@gkatsev, how's that code formatting build step going?) but looks good to me.

File: src/js/player.js [hunk 3]
```diff
@@ -104,7 +104,17 @@ class Player extends Component {
     this.language_ = options['language'] || Options['language'];
 
     // Update Supported Languages
-    this.languages_ = options['languages'] || Options['languages'];
+    if (options['languages']) {
+      // Normalise player option languages to lowercase
+      let languagesToLower = {};
+      Object.getOwnPropertyNames(options['languages']).forEach(function(value,index,array){
```
Comment (gkatsev) [inline src/js/player.js]: space after commas.

**Comment by gkatsev** [comment]
It's coming along slowly. If someone wanted to run it locally, it's doable by installing `videojs-standard` as global and then running `standard` in the videojs folder. Should report all errors. Be aware it will also report warnings, but you can `echo $?` to find out the exit code to see whether it succeeded or failed. Though, error code won't help much until everything gets converted.

**Comment by mister-ben** [comment]
Fixed those and other issues flagged by `videojs-standard`.

**Comment by heff** [comment]
Suggested some changes here: https://github.com/mister-ben/video.js/pull/1

**Pull request closed without merging**